web/src/message_list_view.js

```diff
@@ -398,9 +398,14 @@ export class MessageListView {
         const is_hidden =
             muted_users.is_user_muted(message_container.msg.sender_id) && !is_revealed;
 
+        const is_subscribed = stream_data.is_user_subscribed(
+            message_container.msg.stream_id,
+            people.my_current_user_id(),
+        );
```

Comment:
Hmm how does this interact with DMs? It seems like this check is not in a block limited to stream messages, so this is probably going to happen to return `false`  because `stream_data.is_user_subscribed(undefined, ...)` is probably false?

Reply:
Yeah, DMs were not getting highlighted properly. Changed it to
```js
  const is_direct_message_or_subscribed =
      message_container.msg.type === "private" ||
      stream_data.is_user_subscribed(
          message_container.msg.stream_id,
          people.my_current_user_id(),
      );
```